### PR TITLE
applications: nrf_desktop: Integrate APPLICATION_CONFIG_DIR for sysbuild

### PR DESCRIPTION
**Pull request by mkapala-nordic**
```
Change integrates APPLICATION_CONFIG_DIR for sysbuild. Change also fixes
a few configuration issues observed while testing locally.

Jira: NCSDK-24857
Jira: NCSDK-27858
Jira: NCSDK-27568
```
and
```
applications: nrf_desktop: Remove bootloader-related ops from app confs

After migration to sysbuild, now bootloader-specific Kconfig options
are driven at the sysbuild level, not the main application.
Remove unecessary options that were moved to proper sysbuild*.conf
files.

Jira: NCSDK-27890
```

~Depends on:  https://github.com/nrfconnect/sdk-nrf/pull/15530~

Review only last two commits.

### CODE REVIEW

File: applications/nrf_desktop/Kconfig.sysbuild [hunk 1]
```diff
@@ -1,31 +0,0 @@
-#
```
Comment (MarekPieta) [inline applications/nrf_desktop/Kconfig.sysbuild:1]: Should you appear under both `Co-authored-by` and `Signed-off-by` in the commit message?

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/sysbuild.conf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+################################################################################
+
+SB_CONFIG_SECURE_BOOT_APPCORE=y
+SB_CONFIG_SECURE_BOOT_SIGNING_KEY_FILE="b0_private.pem"
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52840dk_nrf52840/sysbuild.conf:9]: Do we still need to set the `CONFIG_SB_SIGNING_KEY_FILE` in application configuration? Similar for other Kconfig options for B0 bootloader.

File: applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/sysbuild_mcuboot_smp.conf [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+################################################################################
+
+SB_CONFIG_BOOTLOADER_MCUBOOT=y
+SB_CONFIG_BOOT_SIGNATURE_TYPE_RSA=y
+SB_CONFIG_BOOT_SIGNATURE_KEY_FILE="\${APPLICATION_CONFIG_DIR}/images/mcuboot/mcuboot_private.pem"
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/sysbuild_mcuboot_smp.conf:10]: It used to be `CONFIG_BOOT_SIGNATURE_TYPE_ECDSA_P256` here

File: applications/nrf_desktop/configuration/nrf5340dk_nrf5340_cpuapp/sysbuild.conf [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+################################################################################
+
+SB_CONFIG_SECURE_BOOT_APPCORE=y
+SB_CONFIG_SECURE_BOOT_SIGNING_KEY_FILE="b0_private.pem"
+SB_CONFIG_NETCORE_IPC_RADIO=y
+SB_CONFIG_NETCORE_IPC_RADIO_BT_HCI_IPC=y
+
```
Comment (MarekPieta) [inline applications/nrf_desktop/configuration/nrf5340dk_nrf5340_cpuapp/sysbuild.conf:12]: Remove the empty line, please

File: applications/nrf_desktop/sample.yaml [hunk 5]
```diff
@@ -224,11 +175,6 @@ tests:
     extra_args: FILE_SUFFIX=release_fast_pair
                 FP_MODEL_ID=0x8E717D
                 FP_ANTI_SPOOFING_KEY=dZxFzP7X9CcfLPC0apyRkmgsh3n2EbWo9NFNXfVuxAM=
-                SB_CONFIG_BOOTLOADER_MCUBOOT=y
-                SB_CONFIG_MCUBOOT_MODE_DIRECT_XIP=y
-                SB_CONFIG_BT_FAST_PAIR=y
-                SB_CONFIG_PM_EXTERNAL_FLASH_MCUBOOT_SECONDARY=n
-                SB_CONFIG_SECURE_BOOT_APPCORE=n
   applications.nrf_desktop.zrelease_4llpmconn:
```
Comment (MarekPieta) [inline applications/nrf_desktop/sample.yaml:178]: Can we merge entries for `nrf54h20` with remaining targets?

Comment (MarekPieta) [reply applications/nrf_desktop/sample.yaml:178]: Special handling should no longer be needed.

**Review by MarekPieta — CHANGES_REQUESTED** [review]

File: applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/sysbuild_mcuboot_smp.conf (hunk 3 above)
Comment (mkapala-nordic) [reply applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/sysbuild_mcuboot_smp.conf:10]: Oh, it is an only config that has it set, must have missed that. Thanks

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/sysbuild.conf (hunk 2 above)
Comment (mkapala-nordic) [reply applications/nrf_desktop/configuration/nrf52840dk_nrf52840/sysbuild.conf:9]: It is separate ticket that was not addressed yet here -> [NCSDK-27890](https://nordicsemi.atlassian.net/browse/NCSDK-27890)

File: applications/nrf_desktop/Kconfig.sysbuild (hunk 1 above)
Comment (mkapala-nordic) [reply applications/nrf_desktop/Kconfig.sysbuild:1]: `Co-authored-by` is GitHub feature that shows that this commit has multiple authors and `Signed-off-by` is a requirement so the compliance passes.
Probably I could remove `Signed-off-by` and still satisfy compliance but as is also should work.
We will check it after I'll remove the draft status.

Comment (MarekPieta) [reply applications/nrf_desktop/Kconfig.sysbuild:1]: Thanks for clarification

**Review by MarekPieta — APPROVED** [review]
nRF Desktop changes looks good

**Comment by shanthanordic** [comment]
Looks like big change after RC-1 @pdunaj  @kapi-no  Can I push it out of NCs 2.7.0 unless absolutely needed

**Comment by kapi-no** [comment]
> Looks like big change after RC-1 @pdunaj @kapi-no Can I push it out of NCs 2.7.0 unless absolutely needed

If we want to maintain the production quality of the nRF Desktop app, we need this change that is blocked on the @tejlmand PR (dependency indicated in the PR description). Alternatively, we can lower the nRF Desktop quality from production to experimental following the VS code approach. However, going back to the experimental quality for NCS 2.7.0 needs wider discussion from relevant stakeholders.

On a separate note, we had a different arrangement with @carlescufi and @tejlmand, as this change was supposed to be part of NCS 2.7.0.

**Review by tejlmand — APPROVED** [review]

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/prj.conf [hunk 6]
```diff
@@ -106,10 +106,7 @@ CONFIG_ASSERT_LEVEL=2
 CONFIG_DESKTOP_LOG=y
 
 ################################################################################
-# Bootloader Configuration
+# Firmware Info Configuration
```
[the remark this answers is not in the document]
Comment (mkapala-nordic) [reply applications/nrf_desktop/configuration/nrf52840dk_nrf52840/prj.conf]: removed sections and merged with zephyr configs, comment is not necessary imo

File: applications/nrf_desktop/configuration/nrf52840dk_nrf52840/prj_mcuboot_qspi.conf [hunk 7]
```diff
@@ -114,9 +114,6 @@ CONFIG_ASSERT_LEVEL=2
 CONFIG_DESKTOP_LOG=y
 
 ################################################################################
-# Bootloader Configuration
-
-CONFIG_BOOTLOADER_MCUBOOT=y
 
 CONFIG_STREAM_FLASH=y
```
[the remark this answers is not in the document]
Comment (mkapala-nordic) [reply applications/nrf_desktop/configuration/nrf52840dk_nrf52840/prj_mcuboot_qspi.conf]: removed section and merged with zephyr configs

[the remark this answers is not in the document]
Comment (mkapala-nordic) [reply applications/nrf_desktop/configuration/nrf52840dk_nrf52840/prj_mcuboot_qspi.conf]: Follow-up to verify if they are needed: https://nordicsemi.atlassian.net/browse/NCSDK-27950

File: applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/prj_mcuboot_smp.conf [hunk 8]
```diff
@@ -173,9 +173,6 @@ CONFIG_ASSERT_LEVEL=2
 CONFIG_DESKTOP_LOG=y
 
 ################################################################################
-# Bootloader Configuration
-
-CONFIG_BOOTLOADER_MCUBOOT=y
 
 CONFIG_STREAM_FLASH=y
```
[the remark this answers is not in the document]
Comment (mkapala-nordic) [reply applications/nrf_desktop/configuration/nrf52840gmouse_nrf52840/prj_mcuboot_smp.conf]: Follow-up to verify if they are needed: https://nordicsemi.atlassian.net/browse/NCSDK-27950

**Review by MarekPieta — APPROVED** [review]

**Comment by mkapala-nordic** [comment]
Rebased only

**Comment by mkapala-nordic** [comment]
Rebased

**Comment by mkapala-nordic** [comment]
Removed `CONFIG_BOOT_DIRECT_XIP=y` from MCUboot configs as it is propagated from sysbuild (`nrf/sysbuild/CMakeLists.txt:L228`).

**Review by zycz — APPROVED** [review]

**Comment by anangl** [comment]
Rebased after #15530 got merged.

**Review by kapi-no — APPROVED** [review]
Approving it due to imminent RC2.

**Pull request merged**